CMakeLists.txt

```diff
@@ -154,6 +154,8 @@ if(BUILD_TESTING)
     target_link_libraries(test_shared_library ${PROJECT_NAME})
   endif()
 
+  add_library(dummy_compile_test test/test_compile_warnings.cpp)
```

Comment:
Could we make sure to make the deprecation warning an error so that building actually fails if the macro doesn't work?

Reply:
Added "-Werror" for this target

Reply:
ah, `-Werror` doesn't work on Windows. Maybe `/WX`?

Reply:
something like:

```cmake
  if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
      target_compile_options(dummy_compile_test PRIVATE "/WX")
  else()
      target_compile_options(dummy_compile_test PRIVATE "-Werror")
  endif()
```

Reply:
This should be fixed now, if only the CI would run....